crates/burn-import/src/onnx/to_burn.rs

```diff
@@ -1082,12 +1073,27 @@ impl ParsedOnnxGraph {
         UnaryNode::exp(input, output)
     }
 
-    fn expand_conversion(node: Node) -> ExpandNode {
+    fn expand_conversion(mut node: Node) -> ExpandNode {
         let input = TensorType::from(node.inputs.first().unwrap());
-        let output = TensorType::from(node.outputs.first().unwrap());
         let shape = expand_config(&node);
 
-        ExpandNode::new(input, output, shape)
+        // dim_interference left the dim at zero, so it needs to be filled before converting to TensorType:
+        assert_eq!(
+            node.outputs.len(),
+            1,
+            "ExpandNode must have exactly 1 output!"
+        );
+        let mut output_arg = node.outputs.pop().unwrap();
```

Comment:
Instead of taking `mut node: Node` and using `.pop()`, why not use `.first() as before?

Reply:
The Argument in `node.outputs[0]` has `.dim = 0` from `dim_inference` (as the typo-ridden comment also explains 😅), but `TensorType::from` doesn't allow `dim = 0` anymore.
Hence, we need to modify the Arg some way. I saw 3 and thought this the best:
- Clone the Argument. This would work without `mut Node` and with the first, but would unnecessarily clone stuff, which goes against my microoptimizing tendencies
- get an &mut to the Arg within the `outputs`-Vec: still needs `mut Node`, could go with `iter_mut().first()` instead of `.first()`
- `pop()` the single output to a local `mut` Var and edit it there.

It's not a hill I'll die on, so if you prefer another option, I can change it

Reply:
Ahhh you're right, that's my bad. I didn't realize that the `match` expression result was assigned to `output_arg_tensor`.

All good, we can leave it as is.